### PR TITLE
Adds ability to exclude columns from mapping during reveng

### PR DESCRIPTION
**Pull request by sm15455**
The modifications are quite simple. I have added the ExcludedColumns in the TableInformationModel. 
The awkward part is in the ReverseEngineerHandler. I had to replace predefined tables with preselected ones when the names match. This is necessary because the UI serializes from predefined tables and that would overwrite any change made by the user to the efpt.config.json file. If you have a better idea I'm all ears :)

### CODE REVIEW

File: src/GUI/RevEng.Core/VisitorRelationScaffoldingModelFactory.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			else if (_databaseType == DatabaseType.SQLServer)
+			{
+				fullTableName = $"[{table.Schema}].[{table.Name}]";
+			}
+			else
+			{
+				fullTableName = $"{table.Schema}.{table.Name}";
+			}
+			var tableDefinition = _tables.FirstOrDefault(c => c.Name.Equals(fullTableName, StringComparison.OrdinalIgnoreCase));
+			if (tableDefinition != null && tableDefinition?.ExcludedColumns != null) {
```
Comment (ErikEJ) [inline src/GUI/RevEng.Core/VisitorRelationScaffoldingModelFactory.cs:44]: Could this just be: if (tableDefinition?.ExcludedColumns != null) {

File: src/GUI/RevEng.Core/VisitorRelationScaffoldingModelFactory.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+				foreach (var column in tableDefinition?.ExcludedColumns) {
```
Comment (ErikEJ) [inline src/GUI/RevEng.Core/VisitorRelationScaffoldingModelFactory.cs:45]: Please add new line before "{"

File: src/GUI/RevEng.Shared/TableInformationModel.cs [hunk 3]
```diff
@@ -19,6 +20,7 @@ public class TableInformationModel : INotifyPropertyChanged
         private bool _hasPrimaryKey;
         private bool _showKeylessWarning;
         private ObjectType _objectType;
+        private IEnumerable<string> _excludedColumns = new List<string>();
```
Comment (ErikEJ) [inline src/GUI/RevEng.Shared/TableInformationModel.cs:23]: Will this add empty ExcludedColumns arrays to all tables when serialized?

File: src/GUI/RevEng.Shared/TableInformationModel.cs [hunk 4]
```diff
@@ -89,6 +91,21 @@ public ObjectType ObjectType
             }
         }
 
+        /// <summary>
+        /// Gets or sets the object type.
```
Comment (ErikEJ) [inline src/GUI/RevEng.Shared/TableInformationModel.cs:95]: Fix comment

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs [hunk 5]
```diff
@@ -148,10 +148,17 @@ public async System.Threading.Tasks.Task ReverseEngineerCodeFirstAsync(Project p
                     }
                 }
 
+                for (var i = 0; i < predefinedTables.Count; i++)
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs:151]: Suggest you comment this code out with a TODO, and I will investigate once this is merged,

**Review by ErikEJ — CHANGES_REQUESTED** [review]

**Comment by ErikEJ** [comment]
Pls remove the two .zip files, I will update after merge

File: src/GUI/RevEng.Shared/TableInformationModel.cs (hunk 3 above)
Comment (sm15455) [reply src/GUI/RevEng.Shared/TableInformationModel.cs:23]: yes

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs (hunk 5 above)
Comment (sm15455) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs:151]: If I comment this code, the reveng process would overwrite modifications to efpt.config.json file. Is it ok?

**Comment by sm15455** [comment]
Regarding the .zip files, should I revert the commits or should I commit original version?

**Comment by ErikEJ** [comment]
Revert pls

File: src/GUI/RevEng.Shared/TableInformationModel.cs (hunk 3 above)
Comment (ErikEJ) [reply src/GUI/RevEng.Shared/TableInformationModel.cs:23]: Could we avoid that?

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs (hunk 5 above)
Comment (ErikEJ) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs:151]: Yes, I will do some testing when this has been merged

File: src/GUI/RevEng.Shared/TableInformationModel.cs (hunk 3 above)
Comment (sm15455) [reply src/GUI/RevEng.Shared/TableInformationModel.cs]: I removed initialization and added EmitDefaultValues = false to DataMember. It still serializes the null value.
```
{
  ExcludedColumns: null
   ...
}
```

It shouldn't be a problem, but I'll keep investigating.

Comment (ErikEJ) [reply src/GUI/RevEng.Shared/TableInformationModel.cs]: Never mind...

**Review by ErikEJ — APPROVED** [review]

**Comment by ErikEJ** [comment]
LGTM - do you have anything to add?

**Comment by sm15455** [comment]
Nothing to add. I'll work on the UI side on another branch if you think it's ok, but first we would need to discuss about this.

**Comment by ErikEJ** [comment]
Sounds great, let me just think about the UI design first.

**Comment by sm15455** [comment]
Ok, once you merge the code into master I'll open a new issue so we can discuss about it.

**Comment by ErikEJ** [comment]
Thanks for a great contribution! (so far 😄 )

**Pull request merged**